paasta_tools/metrics/metastatus_lib.py

```diff
@@ -244,6 +252,44 @@ def filter_kube_resources(dictionary: Mapping[str, str]) -> Mapping[str, str]:
     return {key: value for (key, value) in dictionary.items() if key in valid_keys}
 
 
+class ResourceParser:
+    @staticmethod
+    def cpus(resources):
+        resources = resources or {}
+        cpu_str = resources.get("cpu", DEFAULT_KUBERNETES_CPU_REQUEST)
+        if cpu_str[-1] == "m":
+            return float(cpu_str[:-1]) / 1000
+        else:
+            return float(cpu_str)
```

Comment:
I'm a little concerned about the edge cases here.
It sounds like there can be things with:
* no resource requests at all
* resources can be None
* it can have an "m" at the end?

Do you you think this is the kind of thing that will change over time and cause unexpected crashes?
If so, can you add a big try/execept so that we don't crash metastatus with unexpected input? I can't imagine it is "that" unexpected, but it is user-input I guess?

Reply:
`parse_size` seems like a bit of a regression from `suffixed_number_value`, since it doesn't handle `Mi` / `Ki`, etc.

Seems pretty easy to make `suffixed_number_value` support `m` though: just make it return a float and add `"m": 1000 ** -1,` to `_IEC_NUMBER_SUFFIXES`.